### ISSUE TITLE
Typescript: error when dynamic import() is used

### ISSUE THREAD

**Issue by Hotell** [prettier/prettier#2177]
**TS version**: 2.4.0
**Prettier version:** 1.4.4

**Code:**

```ts
export default () => {
  return import('./Greeter').then((module)=>module.Greeter)
}
```

**Expected behavior:**

No errors when prettier is run

**Actual behaviour:**

prettier crashes with `SyntaxError: Expression expected.`

command:
`prettier --single-quote --print-width 120 --trailing-comma es5 --no-semi --parser typescript {src,__{tests,mocks}__}/**/*.{ts,tsx} --list-different`

![image](https://user-images.githubusercontent.com/1223799/27256275-435b7bd0-53b0-11e7-9084-df0121b8139f.png)

REPL [link](https://prettier.github.io/prettier/#%7B%22content%22%3A%22const%20getModuleClass%20%3D%20()%20%3D%3E%20import('.%2Fmy-module').then(module%3D%3Emodule.class)%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Atrue%2C%22trailingComma%22%3A%22none%22%2C%22bracketSpacing%22%3Atrue%2C%22jsxBracketSameLine%22%3Afalse%2C%22parser%22%3A%22typescript%22%2C%22semi%22%3Afalse%2C%22useTabs%22%3Afalse%2C%22doc%22%3Afalse%7D%7D)

works with babel/flow

**Comment by Hotell** [comment]
So I guess the issue might be because 1.4.4 has TS 2.3.x as dependency which doesn't support dynamic import()

https://github.com/prettier/prettier/blob/1.4.4/package.json#L47

**Comment by lydell** [comment]
Correct. This is a duplicate of #2117, which was fixed in #2120. Coming in the next version!

**Comment by josephfrazier** [comment]
Yes, this appears to be fixed on master (currently d46b778dba0a7de79aceb9d64c24b799c763fc7a):

```bash
$ pbpaste | prettier --single-quote --print-width 120 --trailing-comma es5 --no-semi --parser typescript
export default () => {
  return import('./Greeter').then(module => module.Greeter)
}
```

As of commit 89971285e4c697705913791c1589c148dbae2dc0, you should be able to install prettier from github, so you don't have to wait for a release:

```bash
yarn add --dev https://github.com/prettier/prettier
```

**Comment by Hotell** [comment]
cheers guys!
